packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1042,12 +1043,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 }
             }
 
+            const pending = this.pendingStateManager.processRemoteMessage(message);
```

Comment:
It's a bit weird from naming and calls that for same message, we can call both processPendingLocalMessage() as well as processRemoteMessage(). Also note that block of code above skips calling into pendingStateManager if it's chunkedOp, should it be the same here? Do we handle correctly chunked local ops?

Reply:
Given both processes return us localMetadata (in different forms), should it be single function call, where more logic is moved into pendingStateManager from here?